### ISSUE TITLE
Doesn't seem to get triggered in rails 4.2.1

### ISSUE THREAD

**Issue by ttcremers** [rails/actionpack-xml_parser#11]
After following the instruction in the Readme it doesn't seem to work. 

I managed to verify that the xml body is contained in request.post_raw but the params hash doesn't contain the hashed version of the xml. As far as I understand the post body should not contain the xml anymore after XmlParamsParser has triggered. 

The xml I'm using is trivial:

```
<site>
  <name>Test</name>
  <domain>example.com</domain>
</site>
```

included and required:

```
config.middleware.insert_after ActionDispatch::ParamsParser, ActionDispatch::XmlParamsParser
```

```
require 'action_dispatch/xml_params_parser'
```

Both in config/application.rb

**Comment by robin850** [comment]
Hi @ttcremers!

Could you give us more details about the problem please ? I'm not able to reproduce this with a Rails 4.2.3 application:

~~~
$ rails new foo && cd foo
$ echo "gem 'actionpack-xml_parser'" >> Gemfile
$ bundle install
~~~

then add the middleware inside `config/application.rb` and with this controller:

~~~ruby
class FooController < ApplicationController
  def bar
    render text: params
  end
end
~~~

and this route:

~~~ruby
match '/bar' => "foo#bar", via: :post
~~~

then running:

~~~
$ curl --data "<site><name>Test</name><domain>example.com</domain></site>" -H "Content-Type:text/xml" http://localhost:3000/bar
{"site"=>{"name"=>"Test", "domain"=>"example.com"}, "controller"=>"foo", "action"=>"bar"}
~~~

Make sure to specify the request's `Content-Type` header to make it work.

**Comment by ttcremers** [comment]
Hi Robin, 

Sure! I was testing on rails version 4.2.1 but I upgrade now to 4.2.3 but I'm still running into the same problem. 

I'm using rspec for my testing 

```ruby
  describe "POST" do
    it "returns http success" do
      site = build(:site) // Factorygirl
      post :create, site.to_xml, { "content_type": "text/xml", format: :xml } 

      expect(response).to have_http_status(:success)
    end
  end
```

The route is simple:

```ruby
  constraints subdomain: 'api' do    
    namespace :api, path: "/" do 
      namespace :v1, defaults: { format: :xml } do 
        resources :sites
      end
    end
  end
```

The action has some extra debugging in it:

```ruby
  def create
    puts "The params hash: #{params.inspect}"
    respond_to do |format|
      format.xml { render xml: Site.create(site_params) }
    end
  end
```

The output of the test including strack trace:

```
11:29:19 - INFO - Running: spec/controllers/api/v1/sites_controller_spec.rb
.The params hash: {"content_type"=>"text/xml", "format"=>"xml", "controller"=>"api/v1/sites", "action"=>"create"}
F

Failures:

  1) Api::V1::SitesController POST returns http success
     Failure/Error: post :create, site.to_xml, { "content_type": "text/xml", format: :xml }
     ActionController::ParameterMissing:
       param is missing or the value is empty: site
     # /Users/thomas/.rvm/gems/ruby-2.2.2/gems/actionpack-4.2.3/lib/action_controller/metal/strong_parameters.rb:251:in `require'
     # ./app/controllers/api/v1/sites_controller.rb:45:in `site_params'
     # ./app/controllers/api/v1/sites_controller.rb:19:in `block (2 levels) in create'
     # /Users/thomas/.rvm/gems/ruby-2.2.2/gems/actionpack-4.2.3/lib/action_controller/metal/mime_responds.rb:215:in `call'
     # /Users/thomas/.rvm/gems/ruby-2.2.2/gems/actionpack-4.2.3/lib/action_controller/metal/mime_responds.rb:215:in `respond_to'
     # ./app/controllers/api/v1/sites_controller.rb:18:in `create'
     # /Users/thomas/.rvm/gems/ruby-2.2.2/gems/actionpack-4.2.3/lib/action_controller/metal/implicit_render.rb:4:in `send_action'
     # /Users/thomas/.rvm/gems/ruby-2.2.2/gems/actionpack-4.2.3/lib/abstract_controller/base.rb:198:in `process_action'
     # /Users/thomas/.rvm/gems/ruby-2.2.2/gems/actionpack-4.2.3/lib/action_controller/metal/rendering.rb:10:in `process_action'
     # /Users/thomas/.rvm/gems/ruby-2.2.2/gems/actionpack-4.2.3/lib/abstract_controller/callbacks.rb:20:in `block in process_action'
     # /Users/thomas/.rvm/gems/ruby-2.2.2/gems/activesupport-4.2.3/lib/active_support/callbacks.rb:115:in `call'
     # /Users/thomas/.rvm/gems/ruby-2.2.2/gems/activesupport-4.2.3/lib/active_support/callbacks.rb:115:in `call'
     # /Users/thomas/.rvm/gems/ruby-2.2.2/gems/activesupport-4.2.3/lib/active_support/callbacks.rb:553:in `block (2 levels) in compile'
     # /Users/thomas/.rvm/gems/ruby-2.2.2/gems/activesupport-4.2.3/lib/active_support/callbacks.rb:503:in `call'
     # /Users/thomas/.rvm/gems/ruby-2.2.2/gems/activesupport-4.2.3/lib/active_support/callbacks.rb:503:in `call'
     # /Users/thomas/.rvm/gems/ruby-2.2.2/gems/activesupport-4.2.3/lib/active_support/callbacks.rb:88:in `run_callbacks'
     # /Users/thomas/.rvm/gems/ruby-2.2.2/gems/actionpack-4.2.3/lib/abstract_controller/callbacks.rb:19:in `process_action'
     # /Users/thomas/.rvm/gems/ruby-2.2.2/gems/actionpack-4.2.3/lib/action_controller/metal/rescue.rb:29:in `process_action'
     # /Users/thomas/.rvm/gems/ruby-2.2.2/gems/actionpack-4.2.3/lib/action_controller/metal/instrumentation.rb:32:in `block in process_action'
     # /Users/thomas/.rvm/gems/ruby-2.2.2/gems/activesupport-4.2.3/lib/active_support/notifications.rb:164:in `block in instrument'
     # /Users/thomas/.rvm/gems/ruby-2.2.2/gems/activesupport-4.2.3/lib/active_support/notifications/instrumenter.rb:20:in `instrument'
     # /Users/thomas/.rvm/gems/ruby-2.2.2/gems/activesupport-4.2.3/lib/active_support/notifications.rb:164:in `instrument'
     # /Users/thomas/.rvm/gems/ruby-2.2.2/gems/actionpack-4.2.3/lib/action_controller/metal/instrumentation.rb:30:in `process_action'
     # /Users/thomas/.rvm/gems/ruby-2.2.2/gems/actionpack-4.2.3/lib/action_controller/metal/params_wrapper.rb:250:in `process_action'
     # /Users/thomas/.rvm/gems/ruby-2.2.2/gems/activerecord-4.2.3/lib/active_record/railties/controller_runtime.rb:18:in `process_action'
     # /Users/thomas/.rvm/gems/ruby-2.2.2/gems/actionpack-4.2.3/lib/abstract_controller/base.rb:137:in `process'
     # /Users/thomas/.rvm/gems/ruby-2.2.2/gems/actionview-4.2.3/lib/action_view/rendering.rb:30:in `process'
     # /Users/thomas/.rvm/gems/ruby-2.2.2/gems/actionpack-4.2.3/lib/action_controller/test_case.rb:632:in `process'
     # /Users/thomas/.rvm/gems/ruby-2.2.2/gems/actionpack-4.2.3/lib/action_controller/test_case.rb:65:in `process'
     # /Users/thomas/.rvm/gems/ruby-2.2.2/gems/actionpack-4.2.3/lib/action_controller/test_case.rb:514:in `post'
     # ./spec/controllers/api/v1/sites_controller_spec.rb:18:in `block (3 levels) in <top (required)>'

Finished in 0.02606 seconds (files took 1.89 seconds to load)
2 examples, 1 failure
``` 

Just to be complete I've used curl as well to do a test as in your example. Here's the command I'm using:

```
curl --data "<site><name>Test</name><domain>example.com</domain></site>" -H "Content-Type:text/xml" http://api.schiema.local:3000/v1/sites.xml```
```

And the response in the log:

```
Started POST "/v1/sites.xml" for 127.0.0.1 at 2015-07-09 11:38:06 +0200
  ActiveRecord::SchemaMigration Load (0.2ms)  SELECT `schema_migrations`.* FROM `schema_migrations`
Processing by Api::V1::SitesController#create as XML
  Parameters: {"subdomain"=>"api"}
The params hash: {"subdomain"=>"api", "format"=>"xml", "controller"=>"api/v1/sites", "action"=>"create"}
Completed 400 Bad Request in 9ms (ActiveRecord: 0.0ms)

ActionController::ParameterMissing (param is missing or the value is empty: site):
  app/controllers/api/v1/sites_controller.rb:45:in `site_params'
  app/controllers/api/v1/sites_controller.rb:19:in `block (2 levels) in create'
  app/controllers/api/v1/sites_controller.rb:18:in `create'


  Rendered /Users/thomas/.rvm/gems/ruby-2.2.2/gems/actionpack-4.2.3/lib/action_dispatch/middleware/templates/rescues/_source.erb (8.5ms)
  Rendered /Users/thomas/.rvm/gems/ruby-2.2.2/gems/actionpack-4.2.3/lib/action_dispatch/middleware/templates/rescues/_trace.html.erb (2.4ms)
  Rendered /Users/thomas/.rvm/gems/ruby-2.2.2/gems/actionpack-4.2.3/lib/action_dispatch/middleware/templates/rescues/_request_and_response.html.erb (0.9ms)
  Rendered /Users/thomas/.rvm/gems/ruby-2.2.2/gems/actionpack-4.2.3/lib/action_dispatch/middleware/templates/rescues/diagnostics.html.erb within rescues/layout (49.3ms)
Cannot render console with content type application/xmlAllowed content types: [#<Mime::Type:0x007fa8e3222368 @synonyms=["application/xhtml+xml"], @symbol=:html, @string="text/html">, #<Mime::Type:0x007fa8e3221eb8 @synonyms=[], @symbol=:text, @string="text/plain">, #<Mime::Type:0x007fa8e320bbe0 @synonyms=[], @symbol=:url_encoded_form, @string="application/x-www-form-urlencoded">]
```

I should mention that I'm also using the Rabl gem. Am I still missing something? If so please let me know and I'll provide it. 

Thanks in advance!

Cheers,
Thomas

**Comment by ttcremers** [comment]
Hi Robin, 

I've seemed to have solved this issue, it looked like middleware just wasn't being loaded so I decided to start a new rails app and, as you did, made an as simple as possible setup. It worked and after writing a test and expanding the setup with my routes etc it also worked. It's a bit mysterious but it seemed that the rails app I was working on earlier had a problem causing it to behave this way.

Thanks for your time. 

Cheers,
Thomas
